src/collective/easyform/browser/fields_listing.pt

```diff
@@ -35,6 +33,11 @@
                   tal:content="group_name"
           >Fieldset name</legend>
 
+          <a id="delete-fieldset-${fieldset_name}" tal:condition="not:repeat/group/start"
```

Comment:
Since these changes are for Plone 6 only I'd recommend to implement the `btn btn-sm btn-danger` classes here too like in `plone.schemaeditor`